src/apps/products/services/product_services.py

```diff
@@ -82,52 +98,53 @@ def get_all_products(
+    product_data = product_input.dict(exclude_unset=True, exclude_none=True)
 
-    if product_data:
-        if product_data.get("name"):
-            product_name_check = session.scalar(
-                select(Product).filter(Product.name == product_input.name).limit(1)
-            )
-            if product_name_check and (product_name_check.id != product_id):
-                raise IsOccupied(Product.__name__, "name", product_input.name)
+    if product_data.get("name"):
+        product_name_check = session.scalar(
+            select(Product).filter(Product.name == product_input.name).limit(1)
+        )
+        if product_name_check and (product_name_check.id != product_id):
+            raise IsOccupied(Product.__name__, "name", product_input.name)
 
-        if product_data.get("category_ids"):
-            incoming_categories = set(product_data["category_ids"])
-            current_categories = set(
-                category.id for category in product_object.categories
-            )
+    if product_data.get("category_ids"):
+        incoming_categories = set(product_data["category_ids"])
+        current_categories = set(
+            category.id for category in product_object.categories
+        )
 
-            if to_delete := current_categories - incoming_categories:
-                session.execute(
-                    delete(category_product_association_table).where(
-                        Category.id.in_(to_delete)
-                    )
+        if to_delete := current_categories - incoming_categories:
+            session.execute(
+                delete(category_product_association_table).where(
+                    Category.id.in_(to_delete)
                 )
+            )
 
-            if to_insert := incoming_categories - current_categories:
-                rows = [
-                    {"product_id": product_id, "category_id": category_id}
-                    for category_id in to_insert
-                ]
-                session.execute(insert(category_product_association_table).values(rows))
-
-            product_data.pop("category_ids")
-
-        statement = (
-            update(Product).filter(Product.id == product_id).values(**product_data)
-        )
+        if to_insert := incoming_categories - current_categories:
+            rows = [
+                {"product_id": product_id, "category_id": category_id}
+                for category_id in to_insert
+            ]
+            session.execute(insert(category_product_association_table).values(rows))
+            
+        
+        product_data.pop("category_ids")
+        
+        if product_data:
+            statement = (
+                update(Product).filter(Product.id == product_id).values(**product_data)
+            )
 
-        session.execute(statement)
-        session.commit()
-        session.refresh(product_object)
+            session.execute(statement)
+            session.commit()
+            session.refresh(product_object)
```

Comment:
change indentation because what if there is no product_data left? Previous changes won't be deployed?